From f0744b7da55c9b5749a1bb6562dd6060ddd46e20 Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Sat, 28 Jan 2017 18:21:11 -0500
Subject: [PATCH 2/3] kernel-yocto: add KBUILD_DEFCONFIG to scc list correctly
 during rebuilds

When using an in-tree defconfig via KBUILD_DEFCONFIG, do_kernel_metadata
only adds WORKDIR/defconfig to the list of sccs if it doesn't already
exist. This is fine on the first build after a cleansstate, but on
subsequent rebuilds (e.g. after a config framgent in the meta layer
changes) the defconfig doesn't get added to sccs or config.queue, or
passed to merge_config.sh

This patch makes sure that KBUILD_DEFCONFIG gets into config.queue
reliably on rebuilds (when WORKDIR/defconfig doesn't exist, or if it
does exist and matches KBUILD_DEFCONFIG)

Signed-off-by: Allen Wild <allenwild93@gmail.com>
---
 meta/classes/kernel-yocto.bbclass | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meta/classes/kernel-yocto.bbclass b/meta/classes/kernel-yocto.bbclass
index 1d447951c4..dd97ab95ce 100644
--- a/meta/classes/kernel-yocto.bbclass
+++ b/meta/classes/kernel-yocto.bbclass
@@ -107,6 +107,8 @@ do_kernel_metadata() {
 				cmp "${WORKDIR}/defconfig" "${S}/arch/${ARCH}/configs/${KBUILD_DEFCONFIG}"
 				if [ $? -ne 0 ]; then
 					bbwarn "defconfig detected in WORKDIR. ${KBUILD_DEFCONFIG} skipped"
+				else
+					sccs="${WORKDIR}/defconfig"
 				fi
 			else
 				cp -f ${S}/arch/${ARCH}/configs/${KBUILD_DEFCONFIG} ${WORKDIR}/defconfig
-- 
2.15.0

